src/auth/sso/validation.ts

```diff
@@ -0,0 +1,51 @@
+/*!
+ * Copyright 2023 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+import * as vscode from 'vscode'
+import { UnknownError } from '../../shared/errors'
+import { AuthType } from '../auth'
+import { isSsoConnection, SsoConnection, hasScopes } from '../connection'
+
+export function validateSsoUrl(auth: AuthType, url: string, requiredScopes?: string[]) {
+    const urlFormatError = validateSsoUrlFormat(url)
```

Comment:
unnecessary indirection. we're already in a function named `validateSsoUrl`. Too much granularity